### PR TITLE
Attach profiler marker fact processor

### PR DESCRIPTION
**Pull request by yasincidem**
#5030

The issue requires ProfilerMarkerFactProcessor([https://github.com/mozilla-mobile/fenix/pull/17841/files](https://github.com/mozilla-mobile/fenix/pull/17841/files.)) to be implemented in this project.

### CODE REVIEW

File: app/src/test/java/org/mozilla/focus/MockitoExt.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus
+
+import org.mockito.ArgumentCaptor
+import org.mockito.Mockito
+
+fun <T> any(): T = Mockito.any()
+
+inline fun <reified T> argumentCaptor(): ArgumentCaptor<T> = ArgumentCaptor.forClass(T::class.java)
```
Comment (jonalmeida) [inline app/src/test/java/org/mozilla/focus/MockitoExt.kt:12]: Let's not duplicate this code. This exists in the Android Components `support-test` module already. Let's include this module as a `testImplementation` dependency instead.

See: 
 - https://github.com/mozilla-mobile/android-components/blob/main/components/support/test/src/main/java/mozilla/components/support/test/KArgumentCaptor.kt
 - https://github.com/mozilla-mobile/android-components/blob/main/components/support/test/src/main/java/mozilla/components/support/test/Matchers.kt

File: app/src/test/java/org/mozilla/focus/telemetry/ProfilerMarkerFactProcessorTest.kt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.never
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.verifyZeroInteractions
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class ProfilerMarkerFactProcessorTest {
+
+    private val profiler: Profiler = mock(Profiler::class.java)
```
Comment (mcomella) [inline app/src/test/java/org/mozilla/focus/telemetry/ProfilerMarkerFactProcessorTest.kt:28]: praise: Nice job converting this from mockk to mockito. :)

**Review by mcomella — APPROVED** [review]
Great, this looks good to me! Thanks for your help, @yasincidem !

**Comment by mcomella** [comment]
I filed https://github.com/mozilla-mobile/android-components/issues/10808 as a follow-up to move the code to our shared library, android-components, and use that shared reference in fenix and focus.

**Pull request merged**